### PR TITLE
Refine exitCode for parent process flow

### PR DESCRIPTION
**Pull request by chenrui333**
Currently the errorMessage could be:
- `Error: 4213-0 already deployed`
- `Error: The following quotas were exceeded: CPUS (quota: 256, used: 256 + needed: 2).`
- Nothing, https://travis-ci.com/meetup/pro-web/jobs/172741859#L706

Any already deployed is good one, the other error code should all stop the process from proceeding further.

### CODE REVIEW

File: src/commands/deployCommands/create.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ module.exports = {
 		}),
 	middlewares: [apiMiddleware],
 	handler: argv =>
-		argv.getDeployApi().then(({ deploy, versions, migrate }) =>
```
Comment (chenrui333) [inline src/commands/deployCommands/create.js]: this is unused object

File: src/commands/deployCommands/create.js [hunk 2]
```diff
@@ -58,10 +58,24 @@ module.exports = {
...
+						// clean up deployed versions and exit
+						return deploy.del().then(() => process.exit(1));
+					} else {
+						console.log(chalk.red(`Stopping deployment: ${error}`));
+						console.log(chalk.red('Cleaning up failed deployment...'));
+
+						let exitCode = 2;
+						// clean up deployed versions and exit
+						return deploy.del().then(() => process.exit(1));
+					}
```
Comment (mmcgahan) [inline src/commands/deployCommands/create.js]: Hmm, it looks like this update doesn't change the behavior of the code - it assigns a local `exitCode` variable, but then doesn't use it on either side of the `if` block, and it doubles the amount of code. I _think_ that the intent is something more like

```js
.catch(error => {
  console.log(chalk.red(`Stopping deployment: ${error}`));
  console.log(chalk.red('Cleaning up failed deployment...'));

  const exitCode = error.includes("already deployed") ? 2 : 1;
  return deploy.del().then(() => process.exit(exitCode));
})
```

Comment (chenrui333) [reply src/commands/deployCommands/create.js]: yeah, you are absolutely right, initially I was thinking to use `if-else` to cover more error codes.

Comment (chenrui333) [reply src/commands/deployCommands/create.js]: I made the acceptable_error and unacceptable_error as constants as well.

Comment (mmcgahan) [reply src/commands/deployCommands/create.js]: Sounds good. If/when we need to cover more error codes, we should probably break out that 'pattern matching' behavior to a separate `getErrorCode(error)` function to encapsulate the logic.

**Comment by chenrui333** [comment]
Thanks @jaclynj, @mmcgahan , merging the PR now!

**Pull request merged**